hydra-node/test/Hydra/NodeSpec.hs

```diff
@@ -10,144 +10,252 @@ import Hydra.API.ClientInput (ClientInput (..))
-
-  it "rotates snapshot leaders" $
-    showLogsOnFailure "NodeSpec" $ \tracer -> do
-      let tx1 = SimpleTx{txSimpleId = 1, txInputs = utxoRefs [2], txOutputs = utxoRefs [4]}
-          sn1 = testSnapshot 1 (utxoRefs [1, 2, 3]) mempty
-          sn2 = testSnapshot 2 (utxoRefs [1, 3, 4]) [1]
-          inputs =
-            inputsToOpenHead
-              <> [ NetworkInput{ttl = defaultTTL, party = alice, message = ReqSn{snapshotNumber = 1, transactionIds = mempty}}
-                 , NetworkInput{ttl = defaultTTL, party = alice, message = AckSn (sign aliceSk sn1) 1}
-                 , NetworkInput{ttl = defaultTTL, party = carol, message = AckSn (sign carolSk sn1) 1}
-                 , NetworkInput{ttl = defaultTTL, party = alice, message = ReqTx{transaction = tx1}}
-                 ]
-
-      node <- createHydraNode bobSk [alice, carol] defaultContestationPeriod inputs
-      (node', getNetworkMessages) <- recordNetwork node
-      runToCompletion tracer node'
-
-      getNetworkMessages `shouldReturn` [AckSn (sign bobSk sn1) 1, ReqSn 2 [1], AckSn (sign bobSk sn2) 2]
-
-  it "processes out-of-order AckSn" $
-    showLogsOnFailure "NodeSpec" $ \tracer -> do
-      let snapshot = testSnapshot 1 (utxoRefs [1, 2, 3]) []
-          sigBob = sign bobSk snapshot
-          sigAlice = sign aliceSk snapshot
-          inputs =
-            inputsToOpenHead
-              <> [ NetworkInput{ttl = defaultTTL, party = bob, message = AckSn{signed = sigBob, snapshotNumber = 1}}
-                 , NetworkInput{ttl = defaultTTL, party = alice, message = ReqSn{snapshotNumber = 1, transactionIds = []}}
-                 ]
-      node <- createHydraNode aliceSk [bob, carol] defaultContestationPeriod inputs
-      (node', getNetworkMessages) <- recordNetwork node
-      runToCompletion tracer node'
-      getNetworkMessages `shouldReturn` [AckSn{signed = sigAlice, snapshotNumber = 1}]
-
-  it "notifies client when postTx throws PostTxError" $
-    showLogsOnFailure "NodeSpec" $ \tracer -> do
-      let inputs = [ClientInput Init]
-      (node, getServerOutputs) <-
-        createHydraNode aliceSk [bob, carol] defaultContestationPeriod inputs
-          >>= throwExceptionOnPostTx NoSeedInput
-          >>= recordServerOutputs
-
-      runToCompletion tracer node
-
-      outputs <- getServerOutputs
-      let isPostTxOnChainFailed = \case
-            PostTxOnChainFailed{postTxError} -> postTxError == NoSeedInput
-            _ -> False
-      any isPostTxOnChainFailed outputs `shouldBe` True
-
-  it "signs snapshot even if it has seen conflicting transactions" $
-    failAfter 1 $
+  -- Set up a hydrate function with fixtures curried
+  let setupHydrate action =
+        showLogsOnFailure "NodeSpec" $ \tracer -> do
+          let testHydrate = hydrate tracer testEnvironment simpleLedger SimpleChainState{slot = ChainSlot 0}
+          action testHydrate
+
+  describe "hydrate" $ do
```

Comment:
NOTE: The diff is not great here, all existing tests un-changed and only moved further down. The `hydrate` and `stepHydraNode` tests groups have been addded.